src/multi.rs

```diff
@@ -147,38 +175,48 @@ macro_rules! impl_tuple {
         }
 
         impl<'lua> FromLuaMulti<'lua> for () {
-            fn from_lua_multi(_: MultiValue, _: Context<'lua>) -> Result<Self> {
+            fn from_lua_multi(_: MultiValue, _: Context<'lua>, _: &mut usize) -> Result<Self> {
                 Ok(())
             }
         }
     );
 
-    ($last:ident $($name:ident)*) => (
-        impl<'lua, $($name,)* $last> ToLuaMulti<'lua> for ($($name,)* $last,)
-            where $($name: ToLua<'lua>,)*
-                  $last: ToLuaMulti<'lua>
+    ($($name:ident)+) => (
+        impl<'lua, $($name),*> ToLuaMulti<'lua> for ($($name,)*)
+            where $($name: ToLuaMulti<'lua>),*
         {
             #[allow(unused_mut)]
             #[allow(non_snake_case)]
             fn to_lua_multi(self, lua: Context<'lua>) -> Result<MultiValue<'lua>> {
-                let ($($name,)* $last,) = self;
+                let ($($name,)*) = self;
 
-                let mut results = $last.to_lua_multi(lua)?;
-                push_reverse!(results, $($name.to_lua(lua)?,)*);
+                let mut results = MultiValue::new();
+                push_reverse!(results, $($name.to_lua_multi(lua)?,)*);
                 Ok(results)
             }
         }
 
-        impl<'lua, $($name,)* $last> FromLuaMulti<'lua> for ($($name,)* $last,)
```

Comment:
Please check this change. I'm guessing `$last` was separate here to allow for `Variadic` last argument.
- `Variadic` can now be in any location but will swallow and error at runtime for inconvertible `Value`s.
  - Should probably be fixed so it works similar to `ab*?c` regex.
- Was the fact that `A` value was placed last in any other way significant?